### PR TITLE
feat: extend organizers with follow functionality

### PR DESCRIPTION
**Pull request by prayag78**
part of #3765 

This PR adds:
- Organizers follow/unfollow functionality
- Notify followers about newly published events
- Add organizer community settings
- Fix some organizer public page UI issues

The follow-up PRs will cover:
- Followed organizers and events to the user dashboard
- Other UI enhancements in the organiser pages
- More organiser community features

Screenshot 2026-06-04 165301

Screenshot 2026-06-06 092842

Screenshot 2026-06-04 171020

Screenshot 2026-06-04 165624

### CODE REVIEW

**Comment by sarafarajnasardi** [comment]
@prayag78 please address the copilot reviews...

**Comment by Aqil-Ahmad** [comment]
@prayag78 only the number of followers are shown for now, right?

File: app/eventyay/base/services/notifications.py [hunk 1]
```diff
@@ -133,3 +135,93 @@ def send_notification_mail(notification: Notification, user: User):
...
+@scopes_disabled()
+def notify_organizer_followers(event_id: int):
+    """
+    Send an email notification to all followers of an organizer when a new event is published.
+
+    This task is dispatched after an event transitions to ``live=True``.
+    It respects the organizer's ``community_follow_enabled`` setting and each
+    follower's locale / timezone.
+    """
+    with transaction.atomic():
```
Comment (Rachit7168) [inline app/eventyay/base/services/notifications.py:150]: Because the transaction.atomic() block ends immediately after this check, the lock on the Event row is released immediately. If this Celery task is triggered twice concurrently (e.g., double-clicks on publish, or task retries), both executions will sequentially acquire the lock, find no LogEntry, and then both will proceed to send duplicate emails to all followers. The actual log entry is only written at the very end of the function, leaving a massive race condition window.

File: app/eventyay/api/views/organizer.py [hunk 2]
```diff
@@ -60,19 +62,63 @@ class OrganizerViewSet(viewsets.ReadOnlyModelViewSet):
...
-                return Organizer.objects.filter(pk__in=self.request.user.teams.values_list('organizer', flat=True))
+                qs = Organizer.objects.filter(pk__in=self.request.user.teams.values_list('organizer', flat=True))
         elif hasattr(self.request.auth, 'organizer_id'):
-            return Organizer.objects.filter(pk=self.request.auth.organizer_id)
+            qs = Organizer.objects.filter(pk=self.request.auth.organizer_id)
         else:
-            return Organizer.objects.filter(pk=self.request.auth.team.organizer_id)
+            qs = Organizer.objects.filter(pk=self.request.auth.team.organizer_id)
+
+        qs = qs.annotate(_follower_count=Count('followers'))
```
Comment (Rachit7168) [inline app/eventyay/api/views/organizer.py]: Using Count() without distinct=True can lead to cartesian products and wildly inaccurate numbers if the base queryset (qs) has other JOINs added to it elsewhere in the code or in future updates. Recommendation: Always use distinct=True when counting reverse relations.

File: app/eventyay/base/services/notifications.py [hunk 3]
```diff
@@ -133,3 +135,93 @@ def send_notification_mail(notification: Notification, user: User):
...
+    followers = (
+        OrganizerFollower.objects.filter(organizer=organizer)
+        .select_related('user')
+        .iterator()
+    )
+
+    tpl_html = get_template('pretixbase/email/organizer_follower_new_event.html')
+    tpl_plain = get_template('pretixbase/email/organizer_follower_new_event.txt')
+
+    for follower in followers:
```
Comment (Rachit7168) [inline app/eventyay/base/services/notifications.py:185]: If an organizer accumulates a massive following (e.g., 20,000+ followers), enqueuing that many tasks synchronously inside a single loop could take a long time, potentially causing the Celery worker to exceed time limits or consume too much memory. Recommendation: This is acceptable for a V1 implementation. For future scaling, consider dispatching "batch" jobs or using Celery's chunk() capabilities.

File: app/eventyay/api/views/organizer.py [hunk 4]
```diff
@@ -60,19 +62,63 @@ class OrganizerViewSet(viewsets.ReadOnlyModelViewSet):
...
+        if not request.user.is_authenticated:
+            return Response({'detail': _('Authentication required.')}, status=status.HTTP_401_UNAUTHORIZED)
+        organizer = self.get_object()
+        deleted, _ = OrganizerFollower.objects.filter(user=request.user, organizer=organizer).delete()
+        return Response({'following': False, 'deleted': deleted > 0}, status=status.HTTP_200_OK)
+
+    @action(detail=True, methods=['get'], url_path='followers')
+    def followers(self, request, *args, **kwargs):
+        organizer = self.get_object()
+        if not organizer.settings.get('community_follow_enabled', as_type=bool, default=True):
```
Comment (Rachit7168) [inline app/eventyay/api/views/organizer.py]: The string 'Following is not enabled for this organizer.' is missing the _() translation wrapper, whereas it is correctly localized in the follow POST action slightly above it.

**Review by Rachit7168 — CHANGES_REQUESTED** [review]
Thanks for working on this.

Please address

**Review by Aqil-Ahmad — CHANGES_REQUESTED** [review]
@prayag78 when disabling the "Allow users to follow this organizer", the number of followers are not shown, even its enabled. Also, when "Allow users to follow this organizer" is disabled the current users who followed can't unfollow. 

I think it would be nice if such edge cases are covered with tests.

I couldn't confirm the email works as i wasn't receiving any, maybe i missed something but not sure.

**Review by Rachit7168 — APPROVED** [review]
LGTM , Approved 

https://github.com/user-attachments/assets/f6e067e1-47c1-4abb-acd2-2965d1afd9e1

https://github.com/user-attachments/assets/53968286-7103-4578-bace-0f071edf7282

**Review by Saksham-Sirohi — APPROVED** [review]
default values are true so these apply unless organizer chooses to disable them
